include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -8790,4 +8790,7 @@ def err_typecheck_non_count_bounds_decl : Error<
 def err_typecheck_non_count_return_bounds : Error<
   "expected %0 to have a pointer or integer return type">;
 
+def err_typecheck_bounds_type_annotation_identifier: Error<
+  "type name cannot have identifier in it">;
```

Comment:
clang points to the problematic type name as part of the error message.    The ability to point at the problematic code is a pretty nice feature of clang's error reporting.